packages/contracts-bedrock/test/scripts/L2Genesis.t.sol

```diff
@@ -0,0 +1,242 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Test } from "forge-std/Test.sol";
+import { stdJson } from "forge-std/StdJson.sol";
+import { console2 as console } from "forge-std/console2.sol";
+import { VmSafe } from "forge-std/Vm.sol";
+
+import { Artifacts } from "scripts/Artifacts.s.sol";
+import { DeployConfig, DEV_ACCOUNT_FUND_AMT } from "scripts/DeployConfig.s.sol";
+import { Executables } from "scripts/Executables.sol";
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { L2GenesisHelpers } from "scripts/libraries/L2GenesisHelpers.sol";
+import { L2Genesis } from "scripts/L2Genesis.s.sol";
+
+/// @notice Reads a `genesis-l2.json` file, parses the `alloc`s, and runs assertions
+///         against each alloc depending on whether it's a precompile, predeploy proxy,
+///         or predeploy implementation.
+contract L2Genesis_Test is Test, Artifacts {
+    DeployConfig public constant cfg =
+        DeployConfig(address(uint160(uint256(keccak256(abi.encode("optimism.deployconfig"))))));
+
+    string internal genesisPath;
+    L2Genesis internal l2Genesis = new L2Genesis();
+
+    struct StorageData {
+        bytes32 key;
+        bytes32 value;
+    }
+
+    /// @notice `balance` and `nonce` are being parsed as `bytes` even though their JSON representations are hex
+    /// strings.
+    ///         This is because Foundry has a limitation around parsing strings as numbers when using `vm.parseJson`,
+    ///         and because we're using `abi.decode` to convert the JSON string, we can't use coersion (i.e.
+    /// `vm.parseJsonUint`)
+    ///         to tell Foundry that the strings are numbers. So instead we treat them as `byte` strings and parse as
+    ///         `uint`s when needed. Additional context: https://github.com/foundry-rs/foundry/issues/3754
+    struct Alloc {
+        address addr;
+        bytes balance;
+        bytes code;
+        bytes nonce;
+        StorageData[] storageData;
+    }
+
+    /// @notice Checks if deploy config file exists at `deployConfigPath`, if it doesn't
+    ///         reverts with warning to run `make devnet-allocs` at monorepo root.
+    ///         Afterwards, checks if genesis file exists at `genesisPath`, and if not
+    ///         runs `L2Genesis.s.sol` script to generate it.
+    function setUp() public override {
+        super.setUp();
+        Artifacts.setUp();
```

Comment:
We just need to call `L2Genesis.setup`?

Reply:
We use `Artifacts.setUp();` to get the `deploymentContext` to be able to read the deploy config